File: app/src/test/java/com/nicholasworkshop/tinklabstest/external/content/ContentServiceIntegrationTest.java
package com.nicholasworkshop.tinklabstest.external.content;

import com.nicholasworkshop.tinklabstest.external.content.model.Story;

import org.junit.Before;
import org.junit.Test;

import java.util.List;

import static org.junit.Assert.assertEquals;

/**
 * Created by nicholas.wong on 2017/05/17.
 */
public class ContentServiceIntegrationTest {

    private ContentService contentService;

    @Before
    public void setUp() throws Exception {
        ContentServiceModule module = new ContentServiceModule();
        contentService = module.contentService();
    }

    @Test
    public void getCityGuide() throws Exception {
        List<Story> storyList = contentService.getCityGuide().blockingFirst();
        assertEquals(102, storyList.size());
    }

    @Test
    public void getShopGuide() throws Exception {
        List<Story> storyList = contentService.getShopGuide().blockingFirst();
        assertEquals(100, storyList.size());
    }

    @Test
    public void getEatGuide() throws Exception {
        List<Story> storyList = contentService.getEatGuide().blockingFirst();
        assertEquals(100, storyList.size());
    }
}

File: app/src/test/kotlin/com/nicholasworkshop/tinklabstest/external/content/ContentServiceIntegrationTest.kt
package com.nicholasworkshop.tinklabstest.external.content

import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test

class ContentServiceIntegrationTest {

    private lateinit var contentService: ContentService

    @Before
    fun setUp() {
        val module = ContentServiceModule()
        contentService = module.contentService()
    }

    @Test
    fun getCityGuide() {
        val storyList = contentService.cityGuide().blockingFirst()
        assertEquals(102, storyList.size.toLong())
    }

    @Test
    fun getShopGuide() {
        val storyList = contentService.shopGuide().blockingFirst()
        assertEquals(100, storyList.size.toLong())
    }

    @Test
    fun getEatGuide() {
        val storyList = contentService.eatGuide().blockingFirst()
        assertEquals(100, storyList.size.toLong())
    }
}